build_system/docker_build.py

```diff
@@ -0,0 +1,77 @@
+
+import re
+import subprocess
+import sys
+
+from cross_build import BuildSystem
+import constants as c
+import build_utils as utils
+
+class DockerBuildSystem(BuildSystem):
+    def clean(self, config):
+        try:
+            self.exec_host_cmd("docker rm -f -v j2v8.$PLATFORM.$ARCH", config)
+        except subprocess.CalledProcessError:
+            return
+
+    def health_check(self, config):
+        try:
+            # general docker availability check
+            self.exec_host_cmd("docker stats --no-stream", config)
+
+            # NOTE: the additional newlines are important for the regex matching
+            version_str = utils.execute_to_str("docker version") + "\n\n"
```

Comment:
can you please commit the fix for:1
1. chmod +x docker/shared/*
2. build_utils.py:40 -> p = subprocess.Popen(cmd, stdout=subprocess.PIPE, universal_newlines=True, shell=True, cwd=cwd)